packages/machine/src/protocol/setup.ts

```diff
@@ -80,13 +80,28 @@ export const SETUP_PROTOCOL: ProtocolExecutionFlow = {
 };
 
 function proposeStateTransition(message: ProtocolMessage, context: Context) {
-  const { multisigAddress } = message.params as SetupParams;
-  const sc = context.stateChannelsMap.get(multisigAddress)!;
-  if (sc === undefined) {
-    console.log("sc keys=", context.stateChannelsMap.keys());
-    throw Error(`no such channel at multisig address ${multisigAddress}`);
+  const {
+    multisigAddress,
+    initiatingAddress,
+    respondingAddress
+  } = message.params as SetupParams;
+
+  if (!context.stateChannelsMap.has(multisigAddress)) {
+    context.stateChannelsMap.set(
+      multisigAddress,
+      new StateChannel(
+        multisigAddress,
+        [initiatingAddress, respondingAddress].sort((a, b) =>
```

Comment:
Can we make this sorting part of the StateChannel constructor?

That way
1) the client doesn't have to keep doing it
2) the sorting implementation becomes standardized

Reply:
I don't think it should be in the constructor actually because it hides some important complexity, but perhaps there could be a _validator_ in the constructor?